pyiceberg/table/__init__.py

```diff
@@ -350,6 +357,241 @@ class RemovePropertiesUpdate(TableUpdate):
     removals: List[str]
 
 
+class TableMetadataUpdateContext:
+    updates: List[TableUpdate]
```

Comment:
I think we should mark this one private for the class:
```suggestion
    _updates: List[TableUpdate]
```